Assets/LeapMotionModules/Attachments/Scripts/Anchors/AnchorableBehaviour.cs

```diff
@@ -0,0 +1,216 @@
+    /// the current anchor inside the AnchorGroup, or null if the object is not currently
+    /// anchored. Setting this property will set the current anchor only if the given
+    /// anchor is a member of the AnchorGroup.
+    /// </summary>
+    public Anchor currentAnchor {
+      get {
+        switch (anchorType) {
+          case AnchorType.SingleAnchor:
+            return anchor;
+          case AnchorType.AnchorGroup: default:
+            return _currentAnchorInGroup;
+        }
+      }
+      set {
+        switch (anchorType) {
+          case AnchorType.SingleAnchor:
+            anchor = value;
+            break;
+          case AnchorType.AnchorGroup: default:
+            if (anchorGroup.ContainsAnchor(value)) {
+              _currentAnchorInGroup = anchor;
+            }
+            else {
+              Debug.LogError("Attempted to set " + this.name + "'s anchor, but the argument anchor is not in "
+                           + "this AnchorableBehaviour's anchor group.");
+              _currentAnchorInGroup = null;
+            }
+            break;
+        }
+      }
+    }
+
+    /// <summary>
+    /// Attempts to re-anchor the object if it is not anchored (i.e. the AnchorableBehaviour
+    /// was disabled).
+    /// 
+    /// If the anchor type is SingleAnchor, the attempt will succeed only if the anchor is
+    /// within its specified range and the anchor is enabled. If the attempt fails, the method
+    /// will return false, although the output anchor will still be provided.
+    /// 
+    /// If the anchor type is AnchorGroup, the attempt will succeed only if there is an
+    /// anchor in the anchorGroup field that is within range and is enabled, otherwise
+    /// the method will return false, and the output anchor will be null.
+    /// </summary>
+    public bool TryToAnchor(out Anchor anchor) {
+      switch (anchorType) {
+        case AnchorType.SingleAnchor:
+          anchor = this.anchor;
+          if (this.enabled) {
+            return true;
+          }
+          else if (anchor.IsWithinRange(this.transform.position)) {
+            this.enabled = true;
+            return true;
+          }
+          else {
+            return false;
+          }
+        case AnchorType.AnchorGroup: default:
+          Anchor toProvide = anchorGroup.FindClosestAnchor(this.transform.position, requireAnchorIsEnabled: true);
+          if (toProvide != null) {
+            this.enabled = true;
+            anchor = toProvide;
+            return true;
+          }
+          else {
+            anchor = null;
+            return false;
+          }
+      }
+    }
+
+    private bool _attachedToAnchor = false;
+
+    void OnValidate() {
+      _isInteractionBehaviourNull = interactionBehaviour == null;
+    }
+
+    void OnEnable() {
+      if (anchorType == AnchorType.SingleAnchor && anchor == null) {
+        this.enabled = false;
```

Comment:
Whew! Just spent a while changing how working with AnchorableBehaviours work so the inspector is less confusing to work with. Now it doesn't get enabled/disabled willy-nilly, the anchorable will just-work if it's enabled and has a valid anchor configuration, otherwise it won't do anything (and will print appropriate errors if it's being set up incorrectly).